### DISCUSSION TITLE
Fronius Smartmeter Simulation for OpenDTU-OnBattery ?

### DISCUSSION THREAD

**Discussion by AloisKlingler** [hoylabs/OpenDTU-OnBattery#835] [Ideas]
Hi,

to have calculation (especially self-consumption) correct in fronius solar web you need to either place a 300€ fronius smart meter before each "foreign" inverter.

so I copy & pasted and bungled around to get a Smartmeter Simulation running on OpenDTU. the outcome is satisfying, it works and can save installation and hardware costs.
With this code parts it is possible to configure one Fronius Gen24 and Fronius Symo an "Smartmeter" as the IP of OpenDTU, so the Fronius Inverter is able to poll current production and overall production continuously like a from a real fronius-smartmeter.

https://github.com/AloisKlingler/OpenDTU-FroniusSM-MB

regrettably I was not able to merge bigger changes of opendtu back, so my project got an (also not well maintained) fork: https://github.com/StoRmtec500/OpenDTU

as OpenDTU officially project does not want to merge underlying modbus implementation (not done by me, was implemented here: https://github.com/ArekKubacki/OpenDTU ) it's a pitty to have no "maintained" OpenDTU, although the initial idea works well.

So, the question, can this be merged to OpenDTU-OnBattery ? Some users from [here](https://www.photovoltaikforum.com/thread/205384-hoymiles-mikrowechselrichter-mit-opendtu-in-solarweb-aufnehmen-ohne-smartmeter/) would be very thankful.

Best regards
Alois

**Comment by cactus-online** [comment]
I like this idea (I use this openDTU version from the beginning)!

**Comment by mucsushifish** [comment]
As we have 2026 now - has this ever been moved forward or implemented? The initial project gets slowly behind the OpenDTU development - while the Modbus implementation works great and is very useful...
